### PR TITLE
Weather widgets

### PR DESCRIPTION
**Pull request by lewy192**
The weather widget potion of the nav bar is complete and fully implemented. Please scrutinise the code and any mistake, bug or performance improvement you come across leave it in the review and I will attend to it as soojn as possible.

Made a few updates since my first commit, let me know how you guys feel about them.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -4,45 +4,86 @@
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>The FlowerPot</title>
-</head>
+    <!-- defer attribute runs the script after the HTML has rendered -->
+    <script src="assets/scripts/script.js" defer></script>
+    <link rel="stylesheet" href="assets/css/reset.css">
+    <link rel="stylesheet" href="assets/css/style.css">
```
Comment (ClareRadtke) [inline index.html]: Note: I believe best practice is to always have your CSS stylesheet linked at the bottom. To ensure your styling is applied over any other styles you might be linking in.

File: demo.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <link rel="stylesheet" href="assets/css/styles.css">
+  <script src="assets/scripts/script.js" defer></script>
```
Comment (ClareRadtke) [inline demo.html:8]: Move JS linking script tag to the last line within your Body tags. It allows the html and styles to load on the page before all of the JS which could cause lag/delays with loading

File: assets/scripts/temp.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+var options = {
+    enableHighAccuracy: true,
+    timeout: 5000,
+    maximumAge: 0
+};
+
+function success(userPosition){
+    let userCoordinatesResponseObject = userPosition.coords;
+    let userCoordinatesObject = {
+        usersLatitude:userCoordinatesResponseObject.latitude,
```
Comment (ClareRadtke) [inline assets/scripts/temp.js:10]: This file looks good -
Only suggestion is a semantic one - you could simplify the usersLatitude/usersLongittude to just usersLat/usersLon

File: assets/scripts/script.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+const weatherApiKey = "7e5922ef7f6bc85e485e53b28667f43a";
+// const weatherApiUrlNormal = `https://api.openweathermap.org/data/2.5/onecall?lat=${lat}&lon=${lon}&exclude=${excludeParams}&appid=${weatherApiKey}&units=metric`
+// const weatherApiUrlRain = `https://api.openweathermap.org/data/2.5/onecall/timemachine?lat=${lat}&lon=${lon}&exclude=${excludeParams}&appid=${weatherApiKey}&units=metric&dt=${yesterdayUnix}`
+
+function updateWeatherWidgets() {
+    var options = {
+        enableHighAccuracy: true,
+        timeout: 5000,
+        maximumAge: 0,
```
Comment (ClareRadtke) [inline assets/scripts/script.js:18]: What does the maximumAge  in options do?

File: assets/scripts/script.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        getWeather(weatherApiUrlRain, true);
+    }
+    function error(err) {
+        // TODO: handle errors
+        console.warn(`ERROR(${err.code}): ${err.message}`);
+    }
+    navigator.geolocation.getCurrentPosition(success, error, options);
+}
+
+function getWeather(weatherApiUrl, rain) {
```
Comment (ClareRadtke) [inline assets/scripts/script.js:44]: Great work on passing the API url in as a parameter to this!

**Review by ClareRadtke — COMMENTED** [review]
Overall this all looks great - my inline comments were mainly suggestions but I did have a question in assets/scripts/script.js. Happy for you to explain this to me via discord if thats easier.

File: index.html (hunk 1 above)
Comment (lewy192) [reply index.html]: Best practice to link the style sheet at the bottom of the head element, correct.
More info can be found here:https://stackoverflow.com/questions/18392418/html-css-put-link-tag-outside-of-the-head#:~:text=7%20Answers&text=Style%20sheets%20are%20linked%20in,document%20from%20the%20top%20again 

the script can be linked at the bottom of the body and this makes it run after the rendering has taken place. However the 'defer' attribute does the same thing, forces the script to wait till rendering has finished before executing. 
I used defer to keep all the linked external files together.

Any futher questions, dont hesitate.

File: demo.html (hunk 2 above)
Comment (lewy192) [reply demo.html:8]: as mentioned in a previous comment, the 'defer' attribute does this and allows to keep all externally linked files in the same place for organisations sake. 

more info may be found here: https://www.w3schools.com/tags/att_script_defer.asp#snhb-main_leaderboard-0:~:text=When%20present%2C%20it%20specifies%20that%20the%20script%20is%20executed%20when%20the%20page%20has%20finished%20parsing.

File: assets/scripts/temp.js (hunk 3 above)
Comment (lewy192) [reply assets/scripts/temp.js:10]: I will remove this file all together as this entire file has been deprecated

File: assets/scripts/script.js (hunk 4 above)
Comment (lewy192) [reply assets/scripts/script.js:18]: this is a parameter of the geolocation api call. 

more info can be found here: https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition#parameters:~:text=maximumAge%3A%20integer%20(milliseconds)%20%7C%20infinity%20%2D%20maximum%20cached%20position%20age.

File: index.html (hunk 1 above)
Comment (ClareRadtke) [reply index.html]: I was unaware of the defer attribute! That's great to know about, thanks for explaining.

**Review by ClareRadtke — APPROVED** [review]
Thank for your responses Lewis!

**Review by ClareRadtke — APPROVED** [review]

**Pull request closed without merging**